src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/action/SearchJqlPreviousReporter.kt

```diff
@@ -0,0 +1,38 @@
+package com.atlassian.performance.tools.jiraactions.api.action
+
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import com.atlassian.performance.tools.jiraactions.api.memories.IssueKeyMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.JqlMemory
+import com.atlassian.performance.tools.jiraactions.api.observation.SearchJqlObservation
+import com.atlassian.performance.tools.jiraactions.api.page.IssueNavigatorPage
+import com.atlassian.performance.tools.jiraactions.jql.BuiltInJQL
+import com.atlassian.performance.tools.jiraactions.api.SEARCH_WITH_JQL_PREV_REPORTER
+import java.util.function.Predicate
+import javax.json.JsonObject
+
+class SearchJqlPreviousReporter(
+    private val jira: WebJira,
+    private val meter: ActionMeter,
+    private val jqlMemory: JqlMemory,
+    private val issueKeyMemory: IssueKeyMemory
+) : Action {
+
+    override fun run() {
+        val jqlQuery = jqlMemory.recallByTag(Predicate { s -> s == BuiltInJQL.REPORTERS.name })!!
```

Comment:
If you don't care about a variable's name, you can use it instead of a single char.

```
Predicate { it == BuiltInJQL.REPORTERS.name }
```